tools/proteore_topgo/topGO.xml

```diff
@@ -0,0 +1,268 @@
+       <param name="textoutput" value="TRUE"/>
+       <param name="plot" value="dotplot,barplot"/>
+       <param name="geneuniverse" value="org.Hs.eg.db"/>
+       <output name="outputtext" file="result.tsv"/>
+     </test>
+   </tests>
+   <help><![CDATA[
+
+
+**Description**
+
+This tool is based on R package topGO. topGO package provides tools for testing GO terms while accounting for the topology of the GO graph. Different test statistics and different methods for eliminating local similarities and dependencies between GO terms can be applied.
+
+This component computes the GO terms representativity of a gene list in one ontology category (Biological Process "BP", Cellular Component "CC", Molecular Function "MF"). This representativity is evaluated in comparison to the background list of all genes/proteins (of the selected species) associated associated with GO terms of the chosen category (BP,CC,MF).
+
+-----
+
+**Input required**
+
+This component works with Ensembl gene IDs (e.g : ENSG0000013618). You can copy/paste these identifiers or supply a tabular file (.csv, .tsv, .txt, .tab)
+and then specifying the column number that contains the ENSG IDs.
+
+-----
+
+**Parameters**
+
+"Species": "Species": the three available species are Homo sapiens, Mus musculus and Rattus norvegicus
+
+"GO terms category": select either Biogical Process (BP)(by default), Cellular Component (CC) or Molecular Function (MF)
+
+"Select GO scoring method: topGO provides the commonly used Fisher test for evaluating which GO terms are over-represented in your gene/protein list; it also provides other GO scoring algorithms (i.e. Elim, Weight01, Parentchild). For the merits of each option and their algorithmic description, please refer to topGO manual:
+https://bioconductor.org/packages/release/bioc/vignettes/topGO/inst/doc/topGO.pdf
+
+"p-value threshold (e.g : 1e-3)": must be in the form of "1e-5" (i.e. 0.00001)
+
+"Multiple testing procedure (p-value adjustment): several FDR procedure for multiple testing and p-value adjustment are available: Holm, Hochberg
+Hommel, Bonferroni, BH (Benjamini-Hochberg), BY (Benjamini-Yekutieli), FDR. Default is BH (most commonly used)
+
+-----
+
+**Output**
+
+Two types of output are available: textual and/or graphical outputs (barplot and/or a dotplot (set by default)).
+
+*Textual output*
+
+The text output lists all the GO-terms that were found significantly enriched according to the specified threshold (p-value).
+
+The different fields are as follow:
+
+- Annotated : number of genes in the selected species that are annotated with the GO-term.
+
+- Significant : number of genes belonging to your input annotated with the GO-term.
+
+- Expected : represents the expected number of interesting genes mapped to the GO term if the interesting genes were randomly distributed over all GO terms.
+
+- p-values : p-value obtained after the test
+
+- ( q-values  : additional column with adjusted pvalues )
+
+-----
+
+.. class:: infomark
```

Comment:
this can be removed ... Galaxy has a build-in feature to expose the dependencies.

Reply:
done